vignettes/fr/datatable-reshape.Rmd

```diff
@@ -0,0 +1,297 @@
+
+```{r}
+DT.m2 = melt(DT, measure = patterns("^dob", "^gender"), value.name = c("dob", "gender"))
+DT.m2
+```
+
+#### - Utilisation de `measure()` pour spécifier `measure.vars` via un séparateur ou un motif
+
+Si, comme dans les données ci-dessus, les colonnes d'entrée à transformer (`melt `) ont des noms réguliers, alors nous pouvons utiliser `measure`, qui permet de spécifier les colonnes à transformer via un séparateur ou une expression régulière. Par exemple, considérons les données `iris`,
+
+```{r}
+(two.iris = data.table(datasets::iris)[c(1,150)])
+```
+
+Les données iris possèdent quatre colonnes numériques avec une structure régulière : d'abord la partie de la fleur, suivie d'un point, puis le type de mesure. Pour spécifier que nous voulons transformer (`melt`) ces quatre colonnes, nous pouvons utiliser `measure` avec `sep="."` ce qui signifie utiliser `strsplit` sur tous les noms de colonnes ; les colonnes qui résultent en un nombre maximum de groupes après division seront utilisées comme `measure.vars` :
+
+```{r}
+melt(two.iris, measure.vars = measure(part, dim, sep="."))
+```
+
+Les deux premiers arguments de `measure` dans le code ci-dessus (`part` et `dim`) sont utilisés pour nommer les colonnes de sortie ; le nombre d'arguments doit être égal au nombre maximum de groupes après division avec `sep`.
+
+Si nous voulons deux colonnes de valeurs, une pour chaque partie, nous pouvons utiliser le mot-clé spécial `value.name`, qui signifie produire une colonne de valeurs pour chaque nom unique trouvé dans ce groupe :
+
+```{r}
+melt(two.iris, measure.vars = measure(value.name, dim, sep="."))
+```
+
+En utilisant le code ci-dessus, nous obtenons une colonne de valeurs par partie de fleur. Si nous voulons une colonne de valeurs pour chaque type de mesure, nous pouvons faire
+
+```{r}
+melt(two.iris, measure.vars = measure(part, value.name, sep="."))
+```
+
+En revenant à l'exemple des données sur les familles et les enfants, nous pouvons voir une utilisation plus complexe de `measure`, impliquant une fonction utilisée pour convertir les valeurs de la chaîne `child` en entiers :
+
+```{r}
+DT.m3 = melt(DT, measure = measure(value.name, child=as.integer, sep="_child"))
+DT.m3
+```
+
+Dans le code ci-dessus, nous avons utilisé `sep="_child"`, ce qui entraîne la transformation des colonnes uniquement si elle contiennent cette chaîne (six noms de colonnes séparés en deux groupes chacun). L'argument `child=as.integer` signifie que le second groupe donnera lieu à une colonne de sortie nommée `child` avec des valeurs définies en appliquant la fonction `as.integer` aux chaînes de caractères de ce groupe.
+
+Enfin, nous considérons un exemple (emprunté au paquet tidyr) où nous devons définir les groupes à l'aide d'une expression régulière plutôt qu'un séparateur.
```

Comment:
this is a really amazing contribution, thanks so much for leading this effort Philippe!
I see "package" translated in French as "paquet" 
I know another translation is "paquetage" (maybe that is Quebeccer regional dialect?)
Either way is fine with me, but I found your French dictionary https://github.com/phgrosjean/rfrench/blob/main/RFrenchDictionary.txt which says that actually "package" (non-translated) is to be preferred, so was "paquet" here a mistake?

Reply:
This is a really tough question. When we wrote the first translation in French of R messages (R v2.2.1), we had to decide how to translate a series of terms. The discussion was hard on "package" because there were several possible translations, but they all have different meanings in computing science. So, we decided **not** to translate it (an R "package" is not really identical to a Debian "paquet/"paquetage").

However, the common use of words is usually the winning one, and many French-speaking people use "paquet". It is also used in R Commander and in the French version of RStudio. So, we tend to consider "package" and "paquet" as synonyms, as both are observed now in roughly equal occurrences.

Do you prefer we choose only one for {data.table}? And in this case, which one? Package or paquet? For me both are OK.

Reply:
Either is ok with me but I would expect that the terms used in the vignette should be consistent with the guidelines in the dictionary.

Reply:
OK, I will change 'paquet' into 'package' everywhere.